### PR TITLE
Refactor IT Engine

### PR DESCRIPTION
**Pull request by dmsolr**
Fixes #ISSUSE_ID.

This PR will miss some features and test scenarios, such as cases run in parallel, until the progress finish.

Changes proposed in this pull request:
- 
-
-

### CODE REVIEW

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/Dockerfile [hunk 1]
```diff
@@ -15,13 +15,25 @@
 # limitations under the License.
 #
 
-FROM openjdk:8-jdk-alpine
+FROM openjdk:8
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/Dockerfile:18]: 8-jdk-alpine is thin enough, can we use it?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/cases/dataset/DataSetLoader.java [hunk 2]
```diff
@@ -52,6 +52,7 @@ public static DataSet load(final String parentPath, final String scenario, final
         }
     }
     
+    @SneakyThrows
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/cases/dataset/DataSetLoader.java:55]: Unnecessary change?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/cases/dataset/DataSetLoader.java [hunk 3]
```diff
@@ -65,6 +66,6 @@ private static String getFile(final String parentPath, final String scenario, fi
         if (new File(result).exists()) {
             return result;
         }
-        throw new IllegalArgumentException(String.format("%s not found, path=%s, scenario=%s, databaseType=%s", dataSetFile, parentPath, scenario, databaseType.getName()));
+        throw new IllegalArgumentException(String.format("%s not found, path=%s, scenario=%s, databaseType=%s", dataSetFile, result, scenario, databaseType.getName()));
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/cases/dataset/DataSetLoader.java:69]: Unnecessary change?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/env/dataset/DataSetEnvironmentManager.java [hunk 4]
```diff
@@ -68,11 +68,12 @@ public DataSetEnvironmentManager(final String dataSetFile, final Map<String, Dat
     
     /**
      * Fill data.
-     * 
-     * @throws SQLException SQL exception
+     *
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/env/dataset/DataSetEnvironmentManager.java:71]: Unnecessary change?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/env/dataset/DataSetEnvironmentManager.java [hunk 5]
```diff
@@ -90,12 +91,7 @@ public void fillData() throws SQLException, ParseException {
             }
             fillDataTasks.add(new InsertTask(actualDataSources.get(dataNode.getDataSourceName()), insertSQL, sqlValueGroups));
         }
-        try {
-            EXECUTOR_SERVICE_MANAGER.getExecutorService().invokeAll(fillDataTasks);
-            // CHECKSTYLE:OFF
-        } catch (final Exception ex) {
-            // CHECKSTYLE:ON
-        }
+        EXECUTOR_SERVICE_MANAGER.getExecutorService().invokeAll(fillDataTasks);
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/env/dataset/DataSetEnvironmentManager.java:94]: Unnecessary change?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/env/dataset/DataSetEnvironmentManager.java [hunk 6]
```diff
@@ -122,7 +118,6 @@ private String generateInsertSQL(final String tableName, final Collection<DataSe
     
     /**
      * Clear data.
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/env/dataset/DataSetEnvironmentManager.java:120]: Unnecessary change?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/annotation/AfterSuite.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+package org.apache.shardingsphere.test.integration.junit.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface AfterSuite {
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/annotation/AfterSuite.java:27]: The class name is too generic, can we consider add prefix to recognize it?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/annotation/BeforeSuite.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+package org.apache.shardingsphere.test.integration.junit.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface BeforeSuite {
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/annotation/BeforeSuite.java:27]: The class name is too generic, can we consider add prefix to recognize it?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/annotation/Conditional.java [hunk 9]
```diff
@@ -15,28 +15,24 @@
...
-/**
- * Parallel runtime strategy.
- */
-@Documented
-@Target(ElementType.TYPE)
+@Target(ElementType.ANNOTATION_TYPE)
 @Retention(RetentionPolicy.RUNTIME)
-@Inherited
-public @interface ParallelRuntimeStrategy {
+public @interface Conditional {
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/annotation/Conditional.java:29]: The class name is too generic, can we consider add prefix to recognize it?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/annotation/Inject.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+package org.apache.shardingsphere.test.integration.junit.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.FIELD)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Inject {
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/annotation/Inject.java:27]: The class name is too generic, can we consider add prefix to recognize it?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/annotation/StorageType.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+package org.apache.shardingsphere.test.integration.junit.annotation;
+
+import lombok.Getter;
+import org.apache.shardingsphere.test.integration.junit.container.H2Container;
+import org.apache.shardingsphere.test.integration.junit.container.MySQLContainer;
+import org.apache.shardingsphere.test.integration.junit.container.StorageContainer;
+
+public enum StorageType {
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/annotation/StorageType.java:25]: Can we consider about using dynamic SPI to instead of static enum?
Just reference OCP.

Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/annotation/StorageType.java:25]: The class name is too generic, can we consider add prefix to recognize it?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/compose/NotSupportedException.java [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.test.integration.junit.compose;
+
+public class NotSupportedException extends Exception {
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/compose/NotSupportedException.java:20]: The class name is too generic, can we consider add prefix to recognize it?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/condition/Condition.java [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.test.integration.junit.condition;
+
+import java.lang.annotation.Annotation;
+
+@FunctionalInterface
+public interface Condition<T extends Annotation> {
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/condition/Condition.java:23]: The class name is too generic, can we consider add prefix to recognize it?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/H2Container.java [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.test.integration.junit.container;
+
+import org.apache.shardingsphere.infra.database.type.dialect.H2DatabaseType;
+
+import javax.sql.DataSource;
+
+public class H2Container extends StorageContainer {
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/H2Container.java:24]: H2 is focus memory, do we need H2 container?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/runner/ShardingSphereITSubCaseRunner.java [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+//    @Override
+//    protected List<FrameworkMethod> computeTestMethods() {
+//        return super.computeTestMethods().stream().filter(e -> {
+//            if (resolver.filter(e)) {
+//                log.warn("{} was ignored.", e.getMethod());
+//                return true;
+//            }
+//            return false;
+//        }).collect(Collectors.toList());
+//    }
```
Comment (terrymanu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/runner/ShardingSphereITSubCaseRunner.java:74]: We need to remove comment codes

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/H2Container.java (hunk 14 above)
Comment (dmsolr) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-fixture/src/test/java/org/apache/shardingsphere/test/integration/junit/container/H2Container.java:24]: We introduce an abstracted storage container to present storage. That is, H2 container is a fake container(need not start a docker container). So that the IT engine can unify manage it.
